mozilla/blurts-server
# Fix not detecting users without scans

# References: 
Jira: MNTOR-2882
Figma: N/A




# Description

In 9dfbb774e04c38d1c725e098c507f4c4d8f7c52a, the function `getOneRepProfileId` was updated to return -1 when a profile wasn't found. This broke various places in our code where we assumed that the user had run a scan already if their OneRep profile ID was a number, and additionally prevented TypeScript from warning us if our code assumes that the user already has a profile ID, since in type-land, -1 is indistinguishable from other numbers.

This changes make the function explicitly return `null` when the user does not have a OneRep profile yet, and updates calling code to actually deal with this. (Presumably, this also fixes a bunch of old issues where we added a type assertion to tell TS that it had returned a number, even though it might have not.)

@mansaj You made the change in https://github.com/mozilla/blurts-server/pull/4119, and @rhelmer, you requested it in https://github.com/mozilla/blurts-server/pull/4116#discussion_r1470456232 - please let me know if I wasn't supposed to have reverted it.

# How to test

Create a new US user and open the dashboard. You should get redirected to the free scan, which didn't happen anymore on `main`.

# Checklist (Definition of Done)
- [ ] I've added a unit test to test for potential regressions of this bug. - No, but type definitions should catch it.
